packages/toolbars-v3/web/src/modals/FloatingPluginMenu/FloatingAddPluginMenu.tsx

```diff
@@ -51,15 +58,17 @@ const FloatingAddPluginMenu: React.FC<Props> = ({ addPluginMenuConfig, plugins }
   };
 
   const toggleAddPluginMenu = () => {
-    modalService?.isModalOpen(MODAL_ID)
-      ? modalService?.closeModal(MODAL_ID)
-      : modalService?.openModal(MODAL_ID, {
+    const isModalOpen = () => modalService.isModalOpen(MODAL_ID);
+    isModalOpen()
```

Comment:
```suggestion
    isModalOpen
```

Please use the isModalOpen property on the component state (so that we can preserve a single source of truth).

Reply:
i have to make a call so it wont be true alll the time